lib/encoder.py

```diff
@@ -59,6 +59,14 @@ def __init__(self, name, values=[]):
     # pylint: disable=W0102
     self.name = name
     self.values = frozenset(values)
+    self.mandatory = False
+
+  def Mandatory(self):
```

Comment:
Normally you would just set option.mandatory (and mandatory would be a property?) but this is fine since you use it as a "trailer" (I wonder what a better term for that is? Builder pattern / fluent APIs)?